ml/backend.go

```diff
@@ -26,19 +26,28 @@ type Backend interface {
 	SystemInfo() string
 }
 
-var backends = make(map[string]func(*os.File) (Backend, error))
+// BackendParams controls how the backend loads and executes models
+type BackendParams struct {
+	NumThreads int // Number of threads to use if running on the CPU
 
-func RegisterBackend(name string, f func(*os.File) (Backend, error)) {
+	MainGPU      int       // Main GPU
+	NumGPULayers int       // Number of layers to offload to GPU
+	TensorSplit  []float32 // Fraction of the model to offload to each GPU
```

Comment:
pedantic but should we make these GoDoc comments above the fields?

```
//MainGPU is the gpu where xyz...
MainGPU int
```

Reply:
Sure